File: typescript-vitest/src/varar/roman-numerals.steps.ts
import { steps } from '@varar/varar'
import { toRoman } from '../roman-numerals.ts'

const { sensor } = steps()

sensor('a decimal and a roman number', (_state, row: { decimal: string; roman: string }) => {
  return { decimal: row.decimal, roman: toRoman(Number(row.decimal)) }
})

File: ruby-rspec/spec/varar/roman_numerals.steps.rb
# frozen_string_literal: true

require 'varar'
require_relative '../../lib/roman_numerals'

steps do
  sensor('a decimal and a roman number') do |_state, row|
    { 'decimal' => row['decimal'], 'roman' => RomanNumerals.to_roman(row['decimal'].to_i) }
  end
end
